### PR TITLE
MAINT: Remove self.assertEqual from testing

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.  Massive, massive PR.

Partially addresses #15990.

### CODE REVIEW

File: pandas/tests/computation/test_eval.py [hunk 1]
```diff
@@ -661,47 +659,47 @@ def test_identical(self):
...
         result = pd.eval('x', engine=self.engine, parser=self.parser)
-        self.assertEqual(result, 1.5)
+        assert result == 1.5
         assert is_scalar(result)
 
         x = False
         result = pd.eval('x', engine=self.engine, parser=self.parser)
-        self.assertEqual(result, False)
+        assert not result
         assert is_scalar(result)
```
Comment (jreback) [inline pandas/tests/computation/test_eval.py]: assert result is False

if the test is actually expecting False and not just Falsey

File: pandas/tests/frame/test_constructors.py [hunk 2]
```diff
@@ -602,66 +601,66 @@ def test_constructor_maskedarray_nonfloat(self):
...
+        assert frame.values.dtype == object
 
         # Check non-masked values
         mat2 = ma.copy(mat)
         mat2[0, 0] = True
         mat2[1, 2] = False
         frame = DataFrame(mat2, columns=['A', 'B', 'C'], index=[1, 2])
-        self.assertEqual(True, frame['A'][1])
-        self.assertEqual(False, frame['C'][2])
+        assert frame['A'][1]
```
Comment (jreback) [inline pandas/tests/frame/test_constructors.py]: is True and is False

**Review by jreback — APPROVED** [review]
a couple of places where I actually want to assert that i am getting a boolean back

use assert is False

File: pandas/tests/computation/test_eval.py (hunk 1 above)
Comment (gfyoung) [inline pandas/tests/computation/test_eval.py]: Done.

File: pandas/tests/frame/test_constructors.py (hunk 2 above)
Comment (gfyoung) [inline pandas/tests/frame/test_constructors.py]: Done.

**Comment by gfyoung** [comment]
Travis is failing for reasons that I don't understand....see <a href="https://travis-ci.org/pandas-dev/pandas/jobs/227165349">here</a>.

**Comment by jreback** [comment]
do those work locally?

you can just revert that module for now and debug / followup after .

**Comment by gfyoung** [comment]
@jreback : Yes, they do.  I ran the code myself.  Also, the `AssertionError` message makes no sense.  It says that `False is False` is not right.  That is absolutely correct.

**Comment by gfyoung** [comment]
@jreback : You know what I can do: I can check `type` (i.e. `assert is_bool(x)`) and see if that works

**Comment by gfyoung** [comment]
@jreback : Workaround succeeded.  Everything is green.

File: pandas/compat/__init__.py [hunk 3]
```diff
@@ -104,6 +104,7 @@ def signature(f):
     map = map
     zip = zip
     filter = filter
+    intern = sys.intern
```
Comment (jreback) [inline pandas/compat/__init__.py]: this is fine
but where is it used?

**Comment by jreback** [comment]
thanks!

File: pandas/compat/__init__.py (hunk 3 above)
Comment (gfyoung) [inline pandas/compat/__init__.py]: I stumbled upon it in a test.

**Pull request merged**